lms/djangoapps/lti_provider/models.py

```diff
@@ -25,25 +30,69 @@ class LtiConsumer(models.Model):
     consumer_secret = models.CharField(max_length=32, unique=True)
 
 
+class OutcomeService(models.Model):
+    """
+    Model for a single outcome service associated with an LTI consumer. Note
+    that a given consumer may have more than one outcome service URL over its
+    lifetime, so we need to store the outcome service separately from the
+    LtiConsumer model.
+
+    An outcome service can be identified in two ways, depending on the
+    information provided by an LTI launch. The ideal way to identify the service
+    is by instance_guid, which should uniquely identify a consumer. However that
+    field is optional in the LTI launch, and so if it is missing we can fall
+    back on the consumer key (which should be created uniquely for each consumer
+    although we don't have a technical way to guarantee that).
+    """
+    lis_outcome_service_url = models.CharField(max_length=255)
+    instance_guid = models.CharField(max_length=255, null=True)
+    consumer_key = models.CharField(max_length=32, db_index=True)
```

Comment:
Not in this case - there can be multiple outcome services for a given LTI consumer (or, more accurately, the outcome service endpoint can change over time).

Although looking at this again, there might be an argument for using a foreign key into the LtiConsumer table, rather than the consumer key. It's largely a style question around normalization - do you have any preferences?